crates/plex-api/src/media_container/server/library.rs

```diff
@@ -0,0 +1,496 @@
+    pub id: u64,
+    pub duration: Option<u64>,
+    pub bitrate: Option<u32>,
+    pub width: Option<u32>,
+    pub height: Option<u32>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
+    pub aspect_ratio: Option<f32>,
+    pub audio_channels: Option<u8>,
+    pub audio_codec: Option<AudioCodec>,
+    pub video_codec: Option<VideoCodec>,
+    pub video_resolution: Option<String>,
+    pub container: Option<ContainerFormat>,
+    pub video_frame_rate: Option<String>,
+    pub audio_profile: Option<String>,
+    pub video_profile: Option<String>,
+    #[serde(rename = "Part")]
+    pub part: Vec<Part>,
+    #[serde(rename = "has64bitOffsets")]
+    pub has_64bit_offsets: Option<bool>,
+    #[serde(default, deserialize_with = "optional_boolish")]
+    pub optimized_for_streaming: Option<bool>,
+    pub display_offset: Option<u64>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+pub struct Tag {
+    pub id: Option<u32>,
+    pub tag: String,
+    pub filter: Option<String>,
+    pub count: Option<u32>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+pub struct Role {
+    pub id: Option<u32>,
+    pub tag: String,
+    pub filter: Option<String>,
+    pub role: Option<String>,
+    pub thumb: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct ParentMetadata {
+    pub parent_key: Option<String>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
+    pub parent_rating_key: Option<u32>,
+    pub parent_guid: Option<String>,
+
+    pub parent_title: Option<String>,
+    pub parent_studio: Option<String>,
+    pub parent_year: Option<u32>,
+    pub parent_content_rating: Option<String>,
+    pub parent_index: Option<u32>,
+
+    pub parent_thumb: Option<String>,
+    pub parent_art: Option<String>,
+    pub parent_theme: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct GrandParentMetadata {
+    pub grandparent_key: Option<String>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
+    pub grandparent_rating_key: Option<u32>,
+    pub grandparent_guid: Option<String>,
+
+    pub grandparent_title: Option<String>,
+    pub grandparent_studio: Option<String>,
+    pub grandparent_year: Option<u32>,
+    pub grandparent_content_rating: Option<String>,
+    pub grandparent_index: Option<u32>,
+
+    pub grandparent_thumb: Option<String>,
+    pub grandparent_art: Option<String>,
+    pub grandparent_theme: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum MetadataType {
```

Comment:
The list is unlikely to change any time soon, but I'd prefer having the `Unknown` variant here, if you don't mind.